### PR TITLE
powerline-go: add fish integration

### PR DESCRIPTION
**Pull request by VergeDX**
### Description
Add fish integration for powerline-go.

### Checklist
- [x] Change is backwards compatible.
- [x] Code formatted with `./format`.
- [x] Code tested through `nix-shell --pure tests -A run.all`.
- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).
- [x] Commit messages are formatted like
```
{component}: {description}
{long description}
```
See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module
  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).
  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: tests/modules/programs/powerline-go/fish.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          ignore-repos = [ "/home/me/project1" "/home/me/project2" ];
+        };
+      };
+    };
+
+    nixpkgs.overlays = [
+      (self: super: {
+        powerline-go = pkgs.writeScriptBin "dummy-pkg" "";
+        zsh = pkgs.writeScriptBin "dummy-pkg" "";
+      })
```
Comment (jonringer) [inline tests/modules/programs/powerline-go/fish.nix:25]: this doesn't look right at all

**Comment by VergeDX** [comment]
@jonringer Changes performed.

**Comment by VergeDX** [comment]
@jonringer Changes performed.
The build phase needed the script of fish, so maybe not overlay for fish is good...

**Comment by rycee** [comment]
Thanks! Merged to master. I fixed so that fish could be replaced by a dummy package.

**Pull request merged**